chain/db/store.go

```diff
@@ -275,6 +275,17 @@ func (cs *ChainStore) persist(b *Block) error {
 		switch txn.UnsignedTx.Payload.Type {
 		case CoinbaseType:
 			coinbase := pl.(*Coinbase)
+			if b.Header.UnsignedHeader.Height != 0 {
+				accSender := cs.States.GetOrNewAccount(BytesToUint160(coinbase.Sender))
+				amountSender := accSender.GetBalance()
+				donation, err := cs.GetDonation()
+				if err != nil {
+					return err
+				}
+				accSender.SetBalance(amountSender - donation.Amount)
```

Comment:
We should probably verify amountSender >= donation.Amount just in case donation calculation has a bug.